public_api_registry.json

```diff
@@ -0,0 +1,380 @@
+  ],
+  "great_expectations.datasource.fluent.spark_datasource.SparkDatasource": [
+    "__init__",
+    "add_dataframe_asset"
+  ],
+  "great_expectations.datasource.fluent.data_asset.path.directory_asset.DirectoryDataAsset": [
+    "add_batch_definition_daily",
+    "add_batch_definition_monthly",
+    "add_batch_definition_whole_directory",
+    "add_batch_definition_yearly"
+  ],
+  "great_expectations.datasource.fluent.spark_filesystem_datasource.SparkFilesystemDatasource": [
+    "__init__"
+  ],
+  "great_expectations.datasource.fluent.spark_dbfs_datasource.SparkDBFSDatasource": [
+    "__init__"
+  ],
+  "great_expectations.datasource.fluent.spark_s3_datasource.SparkS3Datasource": [
+    "__init__"
+  ],
+  "great_expectations.datasource.fluent.spark_google_cloud_storage_datasource.SparkGoogleCloudStorageDatasource": [
+    "__init__"
+  ],
+  "great_expectations.datasource.fluent.spark_azure_blob_storage_datasource.SparkAzureBlobStorageDatasource": [
+    "__init__"
+  ],
+  "great_expectations.datasource.fluent.sqlite_datasource.SqliteDatasource": [
+    "__init__",
+    "add_query_asset",
+    "add_table_asset"
+  ],
+  "great_expectations.datasource.fluent.databricks_sql_datasource.DatabricksSQLDatasource": [
+    "__init__"
+  ],
+  "great_expectations.datasource.fluent.snowflake_datasource.SnowflakeDatasource": [
+    "__init__",
+    "add_table_asset"
+  ],
+  "great_expectations.checkpoint.actions.ValidationAction": [
+    "__init__"
+  ],
+  "great_expectations.checkpoint.actions.SlackNotificationAction": [
+    "__init__"
+  ],
+  "great_expectations.checkpoint.actions.PagerdutyAlertAction": [
+    "__init__"
+  ],
+  "great_expectations.checkpoint.actions.MicrosoftTeamsNotificationAction": [
+    "__init__"
+  ],
+  "great_expectations.checkpoint.actions.OpsgenieAlertAction": [
+    "__init__"
+  ],
+  "great_expectations.checkpoint.actions.EmailAction": [
+    "__init__"
+  ],
+  "great_expectations.checkpoint.actions.UpdateDataDocsAction": [
+    "__init__"
+  ],
+  "great_expectations.checkpoint.actions.SNSNotificationAction": [
+    "__init__"
+  ],
+  "great_expectations.checkpoint.checkpoint.Checkpoint": [
+    "__init__",
+    "run",
+    "save"
+  ],
+  "great_expectations.checkpoint.checkpoint.CheckpointResult": [
+    "describe"
+  ],
+  "great_expectations.core.factory.checkpoint_factory.CheckpointFactory": [
+    "__init__",
+    "add",
+    "all",
+    "delete",
+    "get"
+  ],
+  "great_expectations.core.factory.suite_factory.SuiteFactory": [
+    "__init__",
+    "add",
+    "all",
+    "delete",
+    "get"
+  ],
+  "great_expectations.core.factory.validation_definition_factory.ValidationDefinitionFactory": [
+    "__init__",
+    "add",
+    "all",
+    "delete",
+    "get"
+  ],
+  "great_expectations.data_context.data_context_variables.DataContextVariables": [
+    "__init__",
+    "save"
+  ],
+  "great_expectations.data_context.data_context.abstract_data_context.AbstractDataContext": [
```

Comment:
This is a big one - we don't have docs around this but we've decorated a bunch of methods